processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -53,6 +56,19 @@ public BoundDimFilter(
     this.lowerStrict = (lowerStrict == null) ? false : lowerStrict;
     this.upperStrict = (upperStrict == null) ? false : upperStrict;
     this.alphaNumeric = (alphaNumeric == null) ? false : alphaNumeric;
+    this.extractionFn = extractionFn;
+  }
+
+  public BoundDimFilter(
```

Comment:
Is this constructor needed? If it's only used for tests then usually we modify the tests rather than add another constructor.

(similar comment for all the other filters)